File: web/src/utils/newInline.ts
import type MarkdownNode from "../types/MarkdownNode";

export default function newInline(
	type: string,
	index: number,
	line: number,
	markup: string,
	indent: number,
): MarkdownNode {
	return {
		type,
		block: false,
		index,
		length: 0,
		line,
		markup,
		delimiter: "",
		content: "",
		indent,
		subindent: 0,
		loose: false,
		acceptsContent: false,
		maybeContinuing: false,
		blankAfter: false,
	};
}

File: dotnet/Allmark/Utils/NewInline.cs
namespace Allmark;

using Allmark.Types;

public static partial class Utils
{
    public static MarkdownNode NewInline(string type, int index, int line, string markup, int indent)
    {
        return new MarkdownNode
        {
            Type = type,
            Block = false,
            Index = index,
            Length = 0,
            Line = line,
            Markup = markup,
            Delimiter = "",
            Content = "",
            Indent = indent,
            Subindent = 0,
            AcceptsContent = false,
            MaybeContinuing = false,
            BlankAfter = false,
            Loose = false,
            Children = null
        };
    }
}
